waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/service/impl/PrefixBasedDatabaseMappingService.java

```diff
@@ -239,74 +247,95 @@ public DatabaseMapping databaseMapping(@NotNull String databaseName) {
     return mappings;
   }
 
-  private boolean isWhitelisted(String databasePrefix, String database) {
-    Whitelist whitelist = mappedDbByPrefix.get(databasePrefix);
-    if ((whitelist == null) || whitelist.isEmpty()) {
-      // Accept everything
-      return true;
-    }
-    return whitelist.contains(database);
-  }
+//  private boolean isWhitelisted(String databasePrefix, String database) {
+//    Whitelist whitelist = mappedDbByPrefix.get(databasePrefix);
+//    if ((whitelist == null) || whitelist.isEmpty()) {
+//      // Accept everything
+//      return true;
+//    }
+//    return whitelist.contains(database);
+//  }
 
   @Override
   public PanopticOperationHandler getPanopticOperationHandler() {
     return new PanopticOperationHandler() {
 
+      private void sleepThread() {
+        try {
+          Thread.sleep(2000L);
+        } catch (InterruptedException e) {
+          LOG.info("Execution was cancelled", e.getMessage());
+        }
+      }
+
       @Override
       public List<TableMeta> getTableMeta(String db_patterns, String tbl_patterns, List<String> tbl_types) {
         List<TableMeta> combined = new ArrayList<>();
-        for (Entry<DatabaseMapping, String> mappingWithPattern : databaseMappingsByDbPattern(db_patterns).entrySet()) {
-          try {
-            DatabaseMapping mapping = mappingWithPattern.getKey();
-            String patterns = mappingWithPattern.getValue();
-            List<TableMeta> tables = mapping.getClient().get_table_meta(patterns, tbl_patterns, tbl_types);
-            for (TableMeta tableMeta : tables) {
-              if (isWhitelisted(mapping.getDatabasePrefix(), tableMeta.getDbName())) {
-                combined.add(mapping.transformOutboundTableMeta(tableMeta));
-              }
-            }
-          } catch (TException e) {
-            LOG.warn("Got exception fetching get_table_meta: {}", e.getMessage());
-          }
+        Map<DatabaseMapping, String> databaseMappingsForPattern = databaseMappingsByDbPattern(db_patterns);
+        ExecutorService executorService = Executors.newFixedThreadPool(databaseMappingsForPattern.size());
+        List<Future<List<?>>> futures = new ArrayList<>();
+        String errorMessage = "Got exception fetching get_table_meta: {}";
```

Comment:
Why declare it here as a member var instead of just directly as an arg on line 288?

Reply:
Fair point; I'll do that 👍